package com.fordretain.api.config;

import java.util.List;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import com.fordretain.api.security.AuthenticatedUserArgumentResolver;

@Configuration
public class WebConfig implements WebMvcConfigurer {

	private final AuthenticatedUserArgumentResolver authenticatedUserResolver;

	public WebConfig(AuthenticatedUserArgumentResolver authenticatedUserResolver) {
		this.authenticatedUserResolver = authenticatedUserResolver;
	}

	@Override
	public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
		resolvers.add(authenticatedUserResolver);
	}
}
